tests/decodecorpus.c

```diff
@@ -1121,20 +1148,94 @@ static void initFrame(frame_t* fr)
 }
 
 /* Return the final seed */
-static U32 generateFrame(U32 seed, frame_t* fr)
+static U32 generateFrame(U32 seed, frame_t* fr, dictInfo info)
 {
     /* generate a complete frame */
     DISPLAYLEVEL(1, "frame seed: %u\n", seed);
-
     initFrame(fr);
 
-    writeFrameHeader(&seed, fr);
-    writeBlocks(&seed, fr);
+    writeFrameHeader(&seed, fr, info);
+    writeBlocks(&seed, fr, info);
     writeChecksum(fr);
 
     return seed;
 }
 
+/*_*******************************************************
+*  Dictionary Helper Functions
+*********************************************************/
+/* returns 0 if successful, otherwise returns 1 upon error */
+static int genRandomDict(U32 dictID, U32 seed, size_t dictSize, BYTE* fullDict){
+    /* allocate space for samples */
+    unsigned const numSamples = 4;
+    BYTE samples[5000];
```

Comment:
minor : 5000 bytes is a "relatively" large value for stack allocation.
Granted, on PC/Server, it should not cause troubles,
although we have micro-thread tasks which can be affected.

But just as a good practice, try to avoid that.
If you need a large buffer, `malloc()` it.
And then `free()` it ...